### PR TITLE
Feature/352 dialogue forms

### PR DESCRIPTION
**Pull request by foxleigh81**
## 🧰 Issue
#352 

## 🚀 Overview: 
Builds the three main forms (planning, adjudication and perception) and the form-elements each form requires.

## 🔗 Link to preview

https://serge-review-feature-35-9htnab.herokuapp.com/storybook

## 🤔 Reason: 
These forms are the primary interaction methods in the mapping component.

## 🔨Work carried out:

- [x] Updates build process to use heroku over netlify
- [x] A commit hook automatically runs tests and linting before a push is allowed to avoid broken builds
- [x] Renames old `Dialogue` components to `Form`
- [x] Adds a `label` prop to `input-container`
- [x] Adds new `RCB` component
- [x] Adds new `PlannedRoute` component
- [x] Adds new `Turns` component
- [x] Adds `Planning` Form
- [x] Adds `Adjudication` Form
- [x] Adds `Perception` Form
- [x] Adds 'forces' check to `RCB` component
- [x] Forces now displays an appropriate colour in the RCB component
- [x] Tests pass

## Confirmations

- [x] I have chosen reviewers for my PR.
- [x] I have assigned myself to this PR.
- [x] I have chosen an appropriate label for the PR.
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.
- [x] I confirm that I have checked for required README updates and acted accordingly.

## 📝 Developer Notes:
- Once this PR merges, the storybook link will now be deployed on Heroku using the regular environment link with `/storybook` appended on the end
- This PR introduces push hooks. Once this merges, unit tests and code linting will run locally before a push will be carried out. Failing tests/lints will cause a push to fail.
- `RCB` stands for Radio/Checkbox
- Due to the scope of this task, it is being split into two PR's, the following tasks will be completed in the next PR:

```
- [ ] `Perception` form has access to `MappingContext` and can post data back to the `Mapping` state
- [ ] `Planning` form has access to `MappingContext` and can post data back to the `Mapping` state
- [ ] `Adjudication` form has access to `MappingContext` and can post data back to the `Mapping` state
- [ ] `Adjudication` form displays correct variations, depending on state
- [ ] `Planning` form displays correct variations, depending on state
- [ ] Stories updated/created
```
This means that several things do not work as expected (selected items, forms posting back etc...)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,16 +17,18 @@ matrix:
...
-- yarn lint || travis_terminate 1;
-# start serge
-- yarn start-server &
-# run the tests drop out on failure
-- yarn test || travis_terminate 1;
-# kill open jobs (especially the Serge server running in the background)
-- kill $(jobs -p) || true
+  # build Serge deployment
+  - yarn build
+  # Build the storybook package
```
Comment (IanMayo) [inline .travis.yml:22]: @foxleigh81 - we don't need to build storybook on Travis, do we?

File: packages/components/package.json [hunk 2]
```diff
@@ -15,8 +15,8 @@
     "test:unit": "jest --coverage",
     "lint": "run-p lint:*",
     "lint:ts": "eslint --fix src/**/*.ts*",
-    "lint:styles": "stylelint src/**/*.scss",
-    "build-storybook": "build-storybook -s ./static -c .storybook -o .out"
+    "lint:styles": "stylelint --fix src/**/*.scss",
+    "build-storybook": "build-storybook -s ./static -c .storybook -o ../client/build/storybook && pwd && ls ../client/build/storybook"
```
Comment (IanMayo) [inline packages/components/package.json]: TODO: we still need to remove the pwd & ls from the above line

File: .travis.yml (hunk 1 above)
Comment (foxleigh81) [reply .travis.yml:22]: We don't NEED to but we should. If the storybook build fails, we want to know about it.

File: packages/components/package.json (hunk 2 above)
Comment (foxleigh81) [reply packages/components/package.json]: Ah, I didn't realise it was in there as well. I'll take it out now.

**Review by IanMayo — APPROVED** [review]

**Pull request merged**